esmvalcore/cmor/check.py

```diff
@@ -509,6 +508,7 @@ def _check_coord_bounds(self, cmor, coord, var_name):
                     coord.var_name, var_name)
 
     def _check_time_bounds(self, freq, time):
+        from esmvalcore.preprocessor._time import _get_time_bounds
```

Comment:
Would it be possible to just solve the problem instead of this hacky workaround? I think if you just move the function _get_time_bounds to the the file esmvalcore/cmor/check.py and import it in the preprocessor, things should work just fine. The preprocessor depends on the cmor check module, there should be no need to have the reverse dependency.

Could you also add a unit tests so we do not find out that stuff broke only when building the conda package next time?

Reply:
okay. I thought about that but that func is not really a check function but rather a preprocessor func - the test is there, the test for check.py but for some reason is not run in the test suite? Anyway will have a look now

Reply:
ok I moved around the `_get_time_bounds()` func, even though I don't like it this way either, but heyho. How do you write a test to test for a circular import? Got no clue, the previous case was a function imported in preprocessor from check that was in turn importing the preprocessor, as long as we stay away from `from esmvalcore... import ...` inside any of the cmor modules we're grand

Reply:
> How do you write a test to test for a circular import? 

Just import the module and see if it works.

`import esmvalcore.cmor.check`

raised the circular import exception, so if you add a test that does that, I think we should be good to go. Maybe create a new file and call it `tests/unit/test_cmor_api.py` and have a test there that does the imports of modules documented in the public API? https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.cmor.html

Reply:
that is a good idea! I added that and added top-level imports for all the cmor modules but fine-grained imports only for check module - you reckon that'll do for now?